src/App.tsx

```diff
@@ -109,7 +110,23 @@ export default function App() {
             <Routes>
               {/* Index */}
               <Route path="/" element={<HomePage />} />
-              {/* LoggedIn routes */}
+              {/* LoggedIn routes (user) */}
+              {
+                userState.isLoggedIn ? (
+                  <>
+                    <Route path="/flag/product/:source/:flavor/:barcode" element={<FlagFormPage type_="product" user_id={userState.userName} />} />
+                    <Route path="/flag/image/:source/:flavor/:barcode" element={<FlagFormPage type_="image" user_id={userState.userName} />} />
+                    {/* <Route path="/flag/search/:source/:flavor/:barcode" element={<FlagFormPage type_="search" user_id={userState.userName} />} /> */}
+                  </>
```

Comment:
@alexgarel : I had dynamic routes to flag product and images.
Do you think this is the right way to integrate this into product opener?